brave/browser-laptop
# Refactor about:preferences#payments followups

*This is a follow-up of #6009*

This PR has 6 commits:
1. Polishment of titleBar and walletBar
2. Made the selectors specific on about:preferences#payments to avoid unexpected regressions 
3. Added margin-bottom of the transfer funds button (Redo #6049)
4. Fixed advanced settings modal dialogs on about:preferences#payments
5. Polishment of modal dialog and advanced settings dialog on about:preferences#payments
6. Restyled payment history table

## 1st commit

Polishment of titleBar and walletBar

On about:preferences#payments
- Disabled underline of the anchor link of fundsFAQ (fixes #6083)
- Set font-size of the elements to 14.5px
- Set margin of the elements to 12px
- Set min-width to the select, input, and button
- Set padding, based on the margin of the items inside it

On about:preferences
- Made button size consistent by replacing em with rem (to avoid font-size inconsistency of the buttons inside walletBar) (fixes #6216)
- Set the default margin of the buttons to 0

<img width="838" alt="screenshot 2016-12-10 14 31 04" src="https://cloud.githubusercontent.com/assets/3362943/21071519/19624458-bee6-11e6-8fd0-66a786c59cf0.png">

### Test Plan 1

1. Open about:preferences
2. Make sure the style of about:preferences looks consistent

## 2nd commit

Made the selectors specific on about:preferences#payments to avoid unexpected regressions

- Classes and ids were renamed (fixes #6081)
- Classes were moved under .paymentsContainer, .bitcoinDashboard, and .ledgerTable
- Some elements and classes were removed
- Moved elements above classes/ids
- Moved .verified out of anchor link (the verified icon is no longer clickable) (fix #6225)

See the commit message for more details

Also .paymentHistoryFooter was introduced to fix #6061
<img width="751" alt="screenshot 2016-12-10 14 41 44" src="https://cloud.githubusercontent.com/assets/3362943/21071542/d0974966-bee6-11e6-9c7b-dcc2fc29f84f.png">

### Test Plan 2

1. Make sure all of the selectors listed up in the commit message (except removed ones) can be actually found in the code, especially `preferences.js`
2. Make sure the style of about:preferences#payments is not broken
    - Open about:preferences#payments
    - Toggle off/on switch
    - Click "Advanced Settings..."
    - Click "Backup your wallet" and "Done"
    - Click "Recover your wallet" and "Done"
    - Click "Add funds..."
    - Click "Display QR code"
3. Make sure the verified icon is no longer clickable

## 3rd commit

Added margin-bottom of the transfer funds button (Redo #6049) 
This fixes #6048

<img width="303" alt="screenshot 2016-12-10 14 42 30" src="https://cloud.githubusercontent.com/assets/3362943/21071548/eed1631c-bee6-11e6-800d-8a4fffead94b.png">

### Test Plan 3

1. Open https://jsfiddle.net/LnwtLckc/5/
2. Click "register"
3. Make sure "Transfer BTC" button has the margin-bottom on add funds dialog on about:preferences#payments

## 4th commit

*This commit will be squashed with the 5th commit after review*

Fixed advanced settings modal dialogs on about:preferences#payments

Added white-space:nowrap to recovery keys (fixes #6220)

### Test Plan 4
(n/a; covered by Test Plan 5)

## 5th commit
Polishment of modal dialog and advanced settings dialog on about:preferences#payments

Mockup images by @bradleyrichter: https://github.com/brave/browser-laptop/issues/5719#issuecomment-262431933

Restyled modal dialog under .paymentsContainer
- Made the background of modal dialog consistent
- Changed border-radius of modal dialog (fixes #6223)
- Fixed spacing of the modal dialog header (fixes #6221)
- Set margin-top/margin-bottom to panel inside .board
- Aligned buttons in the footer with flex-end (fixes #6219)
- Aligned fa icons on the payment panel with display:flex (fixes #6222)
- Fixed panelFooter in .addFundsBoard (fixes #5799)
  - Aligned Coinbase icon and message
  - Display the footer

<img width="730" alt="screenshot 2016-12-15 3 32 43" src="https://cloud.githubusercontent.com/assets/3362943/21195457/3429e63a-c277-11e6-9bcb-05050e504513.png">

For US
<img width="730" alt="screenshot 2016-12-15 3 25 51" src="https://cloud.githubusercontent.com/assets/3362943/21195226/3a2c3aa2-c276-11e6-9d04-14c15a302d58.png">

For non-US
<img width="738" alt="screenshot 2016-12-15 3 23 27" src="https://cloud.githubusercontent.com/assets/3362943/21195134/e2f92286-c275-11e6-9014-ee7892ae10e3.png">

Restyled advanced settings dialog
- Designed "hide sites if" options on the dialog (closes #6201, which is a follow-up of #4681)
- Made the margin consistent between the headers/labels and the input/select elements
  - Addresses https://github.com/brave/browser-laptop/pull/6047#issuecomment-265946727

<img width="729" alt="screenshot 2016-12-15 3 33 58" src="https://cloud.githubusercontent.com/assets/3362943/21195506/63961fb0-c277-11e6-963a-f425a1b6008f.png">

## Test Plan 5

For "Restyled modal dialog under .paymentsContainer"
1. Enable Coinbase if not
2. Open about:preferences#payments
3. Click "Add funds..."
4. Make sure the board is well styled
5. Make sure the footer is aligned horizontally
6. Click "Fund with debit/credit"
7. Make sure Coinbase widget is displayed with transparent background
8. Open https://jsfiddle.net/LnwtLckc/5/
9. Click "register" and go back
10. Make sure "Transfer BTC" button is aligned to right

For "Restyled advanced settings dialog"
1. Click "Advanced Settings..."
2. Check the margin between the labels and the select forms
3. Click "Backup your wallet"
4. Check the margin between the labels and the keys
5. Click "Done" and "Recover your wallet"
6. Check the margin between the labels and the input forms

## 6th commit
Restyled payment history table

Closes #6202
Closes #3485

- Removed "pull-right" class from the close button in modalOverlay.less (to confirm 21ece29 of #6009 fixed #5719)
- Added color:@braveMediumOrange to .sectionTitle of .paymentHistory (fixes #6229)
- Added padding-left and padding-right to th and td (fixes #6231)
- Added white-space:nowrap to disable wrap (fixes #6230)
- Added min-width:80px to the button inside the footer (fixes #6232)
- Made sectionTitle consistent
  - Removed ":not(.paymentHistory)" from modalOverlay.less

<img width="736" alt="screenshot 2016-12-15 3 36 37" src="https://cloud.githubusercontent.com/assets/3362943/21195619/e1ad79e8-c277-11e6-8d5d-d5d428b9247e.png">

Test Plan:
1. Disable the ledger
2. run `npm run add-simulated-payment-history` some times
3. Enable the ledger
4. Click "View Payment History..."
5. Click "OK" button
